hawc/apps/assessment/api.py

```diff
@@ -52,6 +52,30 @@ def get_assessment_from_query(request) -> Optional[models.Assessment]:
     return models.Assessment.objects.filter(pk=assessment_id).first()
 
 
+class AssessmentForeignKeyPermissions(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if obj.assessment is None:
+            return bool(request.user and request.user.is_staff)
+        elif request.method in permissions.SAFE_METHODS:
+            return obj.assessment.user_can_view_object(request.user)
+        else:
+            return obj.assessment.user_can_edit_object(request.user)
+
+    def has_permission(self, request, view):
+        if view.action == "list":
+            return bool(request.user and request.user.is_staff)
+        elif view.action == "create":
+            serializer = view.serializer_class(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            assessment = serializer.validated_data.get("assessment")
+            if assessment is None:
+                return bool(request.user and request.user.is_staff)
+            else:
+                return assessment.user_can_edit_object(request.user)
+        else:
+            return True
```

Comment:
return False? safer that way

Reply:
All the other actions not checked by `has_permission` are object specific, so I have to return True to allow `has_object_permission` to be checked. I suppose I could do `return request.user.is_authenticated` as a first pass to make sure the user is logged in?

Reply:
ok, that's fine, but let's add a comment to indicate why we return True